internal/scheduler/scheduling_algo.go

```diff
@@ -91,16 +101,45 @@ func (l *FairSchedulingAlgo) Schedule(
 	overallSchedulerResult := &SchedulerResult{
 		NodeIdByJobId: make(map[string]string),
 	}
-	for _, executor := range accounting.executors {
+
+	executorsToSchedule := accounting.executors
+	slices.SortStableFunc(executorsToSchedule, func(a, b *schedulerobjects.Executor) bool {
+		return strings.Compare(a.Id, b.Id) < 1
+	})
+
+	schedCtx, cancel := context.WithTimeout(ctx, l.maxSchedulingDuration)
+	defer cancel()
+
+	allExecutorsConsidered := false
+	for i, executor := range executorsToSchedule {
+		if schedCtx.Err() != nil {
+			// We've reached the scheduling time limit, exit gracefully
+			break
+		}
+
+		if i+1 == len(executorsToSchedule) {
+			allExecutorsConsidered = true
+		}
+
+		// We sort clusters lexicographically and schedule them all in order - potentially over multiple scheduling rounds
+		// Skip any that have already been considered
+		if executor.Id <= l.previousScheduleClusterId {
+			continue
+		}
+
 		log.Infof("scheduling on %s", executor.Id)
 		schedulerResult, sctx, err := l.scheduleOnExecutor(
```

Comment:
we seem to have ended up with one variable called `sctx` and another called `schedCtx`.   Is it possible to rename at least one of these to distinguish between them!